### PR TITLE
Add Dockerfile for the vale plugin

### PR DESCRIPTION
**Pull request by ericwill**
See eclipse/che#15856

Signed-off-by: Eric Williams <ericwill@redhat.com>

### CODE REVIEW

File: etc/vale.ini [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Example Vale config file (`.vale.ini` or `_vale.ini`)
```
Comment (benoitf) [inline etc/vale.ini:12]: hello, is it possible to add Copyright there as well ?

File: etc/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#!/bin/sh
+#
+# Copyright (c) 2018-2018 Red Hat, Inc.
```
Comment (benoitf) [inline etc/entrypoint.sh]:
```suggestion
# Copyright (c) 2020 Red Hat, Inc.
```

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      echo "Changing permissions on ${f}" && chgrp -R 0 ${f} && \
+      chmod -R g+rwX ${f}; \
+    done
+
+RUN apk --update --no-cache add wget bash && \
+    wget -qO- https://github.com/errata-ai/vale/releases/download/v${VALE_VERSION}/vale_${VALE_VERSION}_Linux_64-bit.tar.gz | tar xvz -C /usr/local/bin && \
+    chmod +x /usr/local/bin/vale
+
+# Future word lists or config files can go here
+RUN mkdir ${HOME}/vale
```
Comment (benoitf) [inline Dockerfile]: this instruction should be merged with the previous RUN instruction

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+ENV VALE_VERSION=1.7.0
+
+RUN mkdir /projects ${HOME} && \
+    # Change permissions to let any arbitrary user
+    for f in "${HOME}" "/etc/passwd" "/projects"; do \
+      echo "Changing permissions on ${f}" && chgrp -R 0 ${f} && \
+      chmod -R g+rwX ${f}; \
+    done
+
+RUN apk --update --no-cache add wget bash && \
```
Comment (benoitf) [inline Dockerfile]: this instruction should be merged with the previous RUN instruction

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    wget -qO- https://github.com/errata-ai/vale/releases/download/v${VALE_VERSION}/vale_${VALE_VERSION}_Linux_64-bit.tar.gz | tar xvz -C /usr/local/bin && \
+    chmod +x /usr/local/bin/vale
```
Comment (benoitf) [inline Dockerfile]: we should probably remove wget (and bash ?) if it's not used anymore after install

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+#   Red Hat, Inc. - initial API and implementation
+
+FROM node:10.16-alpine
```
Comment (benoitf) [inline Dockerfile]: do we need nodejs there ?

File: Dockerfile (hunk 5 above)
Comment (ericwill) [reply Dockerfile]: Do you mean: add them, download/install vale, then remove them?

File: etc/vale.ini (hunk 1 above)
Comment (ericwill) [reply etc/vale.ini:12]: Yes will do.

File: Dockerfile (hunk 5 above)
Comment (benoitf) [reply Dockerfile]: yes. But I'm unsure BTW we need them. Wget is installed by default on alpine. 
And for Bash do we need it ?

Comment (ericwill) [reply Dockerfile]: I just checked and we don't, I left them there by accident for testing purposes.

File: Dockerfile (hunk 6 above)
Comment (ericwill) [reply Dockerfile]: No, I will update to use an alpine-only image.

**Review by benoitf — APPROVED** [review]

**Pull request merged**